src/reader.rs

```diff
@@ -4675,14 +4675,31 @@ impl ReaderData {
         }
     }
 
+    fn should_add_history(&mut self, text: &wstr, parser: &Parser) -> bool {
+        if !function::exists(L!("fish_should_add_history"), parser) {
```

Comment:
> It gets the right-trimmed (WIP) text as the argument.

that sonds fine because that's what will be added, not the original

(Also how about using a mailing list instead of github, to make code review easier)

Reply:
> (Also how about using a mailing list instead of github, to make code review easier)

God no. I don't use email.